### PR TITLE
Ensure task mail respects `[scheduler][mail]to/from`

### LINKED ISSUE #5976: mail: [scheduler][mail]to is not being used as the default for [runtime][root][mail]to
According to the config change proposal, the scheduler's `[mail]to` is supposed to act at the default for the `[runtime]` equivalent.

https://cylc.github.io/cylc-admin/proposal-config-changes.html#cylceventsmail----schedulermail

So, in this example the failed task mail should be sent to `me@example.com`:

```ini
[scheduler]
    [[mail]]
        to = me@example.com

[scheduling]
    [[graph]]
        R1 = foo

[runtime]
    [[root]]
        [[[events]]]
            mail events = failed

    [[foo]]
        script = false
```

But it doesn't look like that's happening.

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #5976

Plus tidy some cruft I spotted along the way (sorry, couldn't resist)

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] Tests are included
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] No docs entry needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/unit/test_task_events_mgr_2.py
Comment (MetRonnie) [inline tests/unit/test_task_events_mgr_2.py:1]: Moved this into `test_task_events_mgr.py`. Not sure why there was ever a "2" file?

File: cylc/flow/task_events_mgr.py [hunk 1]
```diff
@@ -429,23 +429,23 @@ class TaskEventsManager():
...
+    mail_smtp: Optional[str] = None
+    mail_footer: Optional[str] = None
+
     def __init__(
         self, workflow, proc_pool, workflow_db_mgr, broadcast_mgr,
         xtrigger_mgr, data_store_mgr, timestamp, bad_hosts,
         reset_inactivity_timer_func
     ):
         self.workflow = workflow
-        self.workflow_url = None
```
Comment (MetRonnie) [inline cylc/flow/task_events_mgr.py:435]: This was unused

**Review by hjoliver — APPROVED** [review]
LGTM

**Review by wxtim — APPROVED** [review]
I'm satisfied this works. Have double checked that it doesn't break Cylc Review.

**Pull request merged**